modules/core/src/bitgo.ts

```diff
@@ -1442,159 +1298,139 @@ export class BitGo {
           throw new Error('Keychain needs encryptedXprv property');
         }
 
-        const responseDetails = self.handleTokenIssuance(response.body, password);
-        self._token = responseDetails.token;
-        self._ecdhXprv = responseDetails.ecdhXprv;
+        const responseDetails = this.handleTokenIssuance(response.body, password);
+        this._token = responseDetails.token;
+        this._ecdhXprv = responseDetails.ecdhXprv;
 
         // verify the response's authenticity
-        request.verifyResponse(response);
+        verifyResponse(this, responseDetails.token, 'post', request, response);
 
         // add the remaining component for easier access
-        response.body.access_token = self._token;
+        response.body.access_token = this._token;
       }
 
-      return response;
-    }).call(this)
-      .then(handleResponseResult(), handleResponseError)
-      .nodeify(callback);
+      return handleResponseResult<any>()(response);
+    } catch (e) {
+      handleResponseError(e);
+    }
   }
 
   /**
    * @param params
    * - operatingSystem: one of ios, android
    * - pushToken: hex-formatted token for the respective native push notification service
-   * @param callback
    * @returns {*}
    * @deprecated
    */
-  registerPushToken(params, callback?: NodeCallback<any>): Bluebird<any> {
+  async registerPushToken(params: RegisterPushTokenOptions): Promise<any> {
     params = params || {};
-    common.validateParams(params, ['pushToken', 'operatingSystem'], [], callback);
+    common.validateParams(params, ['pushToken', 'operatingSystem'], []);
 
     if (!this._token) {
       // this device has to be registered to an extensible session
-      return this.reject('not logged in', callback);
+      throw new Error('not logged in');
     }
 
     const postParams = _.pick(params, ['pushToken', 'operatingSystem']);
 
     return this.post(this.url('/devices'))
       .send(postParams)
-      .result()
-      .nodeify(callback);
+      .result();
   }
 
-
   /**
-   *
    * @param params
    * - pushVerificationToken: the token received via push notification to confirm the device's mobility
-   * @param callback
    * @deprecated
    */
-  verifyPushToken(params: VerifyPushTokenOptions, callback?: NodeCallback<any>): Bluebird<any> {
-    const self = this;
-    return co(function *() {
-      if (!_.isObject(params)) {
-        throw new Error('required object params');
-      }
+  verifyPushToken(params: VerifyPushTokenOptions): any {
```

Comment:
This method will return a promise, right?

Reply:
Yes it will, good catch. I'll fix this in the follow up PR